# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaManager.h
# Commit: 396d9243d417
# Full Hash: 396d9243d4172aef258b24a09cf2a2574d86f87a
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-09 09:46:37
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add a dedicated method for clearing storages of a specific client; r=dom-storage-reviewers,jari
#   
#   Client storages can be currently cleared by using the optional argument of
#   nsIQuotaManagerService::ClearStoragesForPrincipal. However, supporting clearing
#   storages for both a principal and a client is not ideal for several reasons.
# ==============================================================================

diff -r fe4a050ac4da -r 396d9243d417 dom/quota/QuotaManager.h
--- a/dom/quota/QuotaManager.h	Wed Oct 09 05:34:35 2024 +0000
+++ b/dom/quota/QuotaManager.h	Wed Oct 09 05:34:36 2024 +0000
@@ -441,6 +441,10 @@
       const PrincipalInfo& aPrincipalInfo,
       const Maybe<Client::Type>& aClientType);
 
+  RefPtr<BoolPromise> ClearStoragesForClient(
+      Maybe<PersistenceType> aPersistenceType,
+      const PrincipalInfo& aPrincipalInfo, Client::Type aClientType);
+
   RefPtr<BoolPromise> ClearStoragesForOriginPrefix(
       const Maybe<PersistenceType>& aPersistenceType,
       const PrincipalInfo& aPrincipalInfo);